src/Test/UnitTests/objectives.jl

```diff
@@ -314,6 +314,50 @@ function solve_qp_edge_cases(model::MOI.ModelLike, config::TestConfig)
 end
 unittests["solve_qp_edge_cases"] = solve_qp_edge_cases
 
+"""
+    solve_qp_zero_offdiag(model::MOI.ModelLike, config::TestConfig)
+
+Test quadratic program with a zero off-diagonal term.
+
+If `config.solve=true` confirm that it solves correctly.
+"""
+function solve_qp_zero_offdiag(model::MOI.ModelLike, config::TestConfig)
+    MOI.empty!(model)
+    x = MOI.add_variables(model, 2)
+    MOI.set(model, MOI.ObjectiveSense(), MOI.MIN_SENSE)
+    vc1 = MOI.add_constraint(
+        model,
+        MOI.SingleVariable(x[1]),
+        MOI.GreaterThan(1.0),
+    )
+    @test vc1.value == x[1].value
+    vc2 = MOI.add_constraint(
+        model,
+        MOI.SingleVariable(x[2]),
+        MOI.GreaterThan(2.0),
+    )
+    @test vc2.value == x[2].value
+    MOI.set(
+        model,
+        MOI.ObjectiveFunction{MOI.ScalarQuadraticFunction{Float64}}(),
+        MOI.ScalarQuadraticFunction(
+            MOI.ScalarAffineTerm{Float64}[],  # affine terms
+            MOI.ScalarQuadraticTerm.(
+                [2.0, 0.0, 2.0],
+                [x[1], x[1], x[2]],
+                [x[1], x[2], x[2]],
+            ),  # quad
+            0.0,  # constant
+        ),
+    )
+    test_model_solution(
+        model,
+        config;
+        objective_value = 5.0,
+        variable_primal = [(x[1], 1.0), (x[2], 2.0)],
+    )
+end
```

Comment:
Needs
```
unittests["solve_qp_zero_offdiag"] = solve_qp_zero_offdiag
```

This could go in `solve_qp_edge_cases` so it doesn't break the tests of every solver who doesn't support QP? Or add a check for `supports`?

Reply:
I tried adding it to `edge_case` but the bug in OSQP only occurs at `copy_to`, not if you do `MOI.set` so it didn't break OSQP.